### PR TITLE
Add HL_HEX_TIMING and HL_HEX_SIM_STATS support

### PR DESCRIPTION
**Pull request by ronlieb**

### CODE REVIEW

**Comment by pranavb-ca** [comment]
LGTM.

File: src/runtime/hexagon_remote/sim_host.cpp [hunk 1]
```diff
@@ -29,6 +29,15 @@ int init_sim() {
         return -1;
     }
 
+    const char *Timing = getenv("HL_HEX_TIMING");
```
Comment (dsharletg) [inline src/runtime/hexagon_remote/sim_host.cpp]: Local variables should be lower case

File: src/runtime/hexagon_remote/sim_host.cpp [hunk 2]
```diff
@@ -309,6 +317,18 @@ int halide_hexagon_remote_run(handle_t module_ptr, handle_t function,
         printf("Hexagon simulator executed function 0x%x in %d cycles\n", function, cycles);
     }
 
+    if (getenv("HL_HEX_SIM_STATS")) {
+        HEXAPI_Status status = HEX_STAT_SUCCESS;
```
Comment (dsharletg) [inline src/runtime/hexagon_remote/sim_host.cpp]: Can you declare this on the line below where it's used? HEXAPI_Status status = ...

File: src/runtime/hexagon_remote/sim_host.cpp [hunk 3]
```diff
@@ -309,6 +317,18 @@ int halide_hexagon_remote_run(handle_t module_ptr, handle_t function,
...
+        char Buf[4096];
+        status = sim->EmitPerfStatistics(0,0,0,0,Buf, sizeof(Buf));
```
Comment (dsharletg) [inline src/runtime/hexagon_remote/sim_host.cpp]: Should be spaces in the argument list.

File: src/runtime/hexagon_remote/sim_host.cpp [hunk 4]
```diff
@@ -309,6 +317,18 @@ int halide_hexagon_remote_run(handle_t module_ptr, handle_t function,
...
+        if (status != HEX_STAT_SUCCESS) {
+            printf("HexagonWrapper::EmitStatistics failed: %d\n", status);
+        } else {
+            // Print out the full stats
+            printf("%s\n", Buf);
```
Comment (dsharletg) [inline src/runtime/hexagon_remote/sim_host.cpp]: Is there a way to reset the stats prior to running the pipeline? This is going to print out the stats from all past simulated code, even if many pipelines have run previously.

**Comment by ronlieb** [comment]
i think there may be a way to reset stats between runs, looking into it more.

**Comment by ronlieb** [comment]
i looked around and tried ResetPmu(), nope. I tried calling EmitPerfStatistics and passing in beginning and ending times, but that did not seem to work.i asked our simulator expert and he was not aware of a way to reset it.

so we can remove the functionality.

or we can leave it in, and its off by default.
And we just need to know its cumulative.

preference ?

**Comment by dsharletg** [comment]
In that case, maybe it should be emitted at the end of the program, instead of after each pipeline run?

**Comment by ronlieb** [comment]
ok, revised

**Comment by dsharletg** [comment]
The code looks good, thanks for the fixes. One last thing: can you recompile the simulator remote runtimes that are checked in and add them to the branch? I know it's annoying to have compiled binaries checked in, but I'm not sure how to avoid it for now.

**Comment by ronlieb** [comment]
sure think will do

**Comment by ronlieb** [comment]
so you want these 5 committed ?
#       modified:   src/runtime/hexagon_remote/arm-32-android/libhalide_hexagon_host.so
#       modified:   src/runtime/hexagon_remote/arm-64-android/libhalide_hexagon_host.so
#       modified:   src/runtime/hexagon_remote/host/libhalide_hexagon_host.so
#       modified:   src/runtime/hexagon_remote/v60/hexagon_sim_remote
#       modified:   src/runtime/hexagon_remote/v60/libhalide_hexagon_remote_skel.so

**Comment by dsharletg** [comment]
I was only expecting this one to change...

    src/runtime/hexagon_remote/host/libhalide_hexagon_host.so

**Comment by ronlieb** [comment]
done

**Pull request merged**